internal/benchrunner/runners/system/metrics.go

```diff
@@ -126,24 +153,98 @@ func (c *collector) collect() {
 		m.dsMetrics = dsstats
 	}
 
-	c.collectedMetrics = append(c.collectedMetrics, m)
+	return m
+}
+
+func (c *collector) publish(events [][]byte) {
+	if c.msapi == nil {
+		return
+	}
+	for _, e := range events {
+		reqBody := bytes.NewReader(e)
+		resp, err := c.msapi.Index(c.indexName(), reqBody)
+		if err != nil {
+			logger.Debugf("error indexing event: %v", err)
+			continue
+		}
+
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			logger.Errorf("failed to read index response body: %v", err)
+		}
+
+		if resp.StatusCode != 201 {
+			logger.Errorf("error indexing event (%d): %s: %v", resp.StatusCode, resp.Status(), elasticsearch.NewError(body))
+		}
+	}
+}
+
+func (c *collector) createMetricsIndex() {
+	if c.msapi == nil {
+		return
+	}
+	reader := bytes.NewReader(
+		[]byte(`{
+			"settings": {
+				"number_of_replicas": 0
+			},
+			"mappings": {
```

Comment:
Should we have a more specific mapping? Are all possible fields integer numbers and strings?

Reply:
all values should be properly mapped with this one as there are only integers, strings and booleans in the metrics